target/linux/ath79/dts/ar9132_buffalo_wzr-hp-g300nh.dts

```diff
@@ -0,0 +1,257 @@
+			label = "movie_on";
+			linux,code = <BTN_4>;
+			gpios = <&gpio2 7 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+	};
+
+	flash@1e000000 {
+		compatible = "cfi-flash";
+		reg = <0x1e000000 0x2000000>;
+		bank-width = <2>;
+		device-width = <2>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0000000 0x0040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x0040000 0x0020000>;
+				read-only;
+			};
+
+			partition@60000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x0060000 0x1f60000>;
+			};
+
+			partition@1fc0000 {
+				label = "user_property";
+				reg = <0x1fc0000 0x0020000>;
+				read-only;
+			};
+
+			art: partition@1fe0000 {
+				label = "art";
+				reg = <0x1fe0000 0x020000>;
+				read-only;
+			};
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		usb {
+			label = "buffalo:blue:usb";
+			gpios = <&gpio 0 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&hub_port>;
+			linux,default-trigger = "usbport";
+		};
+
+		led_diag: diag {
+			label = "buffalo:red:diag";
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+		};
+
+		wireless {
+			label = "buffalo:green:wireless";
+			gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		led_security: security {
+			label = "buffalo:amber:security";
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+		};
+
+		router {
+			label = "buffalo:green:router";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	switch {
+		/* The device may contain either of the switches */
+		compatible = "realtek,rtl8366rb", "realtek,rtl8366s";
+		gpio-sda = <&gpio 19 GPIO_ACTIVE_HIGH>;
+		gpio-sck = <&gpio 20 GPIO_ACTIVE_HIGH>;
+		mii-bus = <&mdio0>;
+
+		mdio-bus {
+			status = "okay";
+
+			#address-cells = <1>;
+			#size-cells = <0>;
+			phy-mask = <0x10>;
+
+			phy4: ethernet-phy@4 {
+				reg = <4>;
+				phy-mode = "rgmii";
+			};
+		};
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	pll-data = <0x1e000100 0x13000a44 0x00441099>;
+	phy-mode = "rgmii";
+	mtd-mac-address = <&art 0x1120c>;
+
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+};
+
+&eth1 {
+	status = "okay";
+
+	compatible = "qca,ar9130-eth", "syscon";
+	reg = <0x1a000000 0x200 0x18070004 0x4>;
+	pll-reg = <0x4 0x18 22>;
+	pll-handle = <&pll>;
+	pll-data = <0x1e000100 0x13000a44 0x00441099>;
+
+	resets = <&rst 13>;
+	reset-names = "mac";
+
+	qca,mac-idx = <1>;
+
+	phy-mode = "rgmii";
+	phy-handle = <&phy4>;
+
+	mtd-mac-address = <&art 0x1120c>;
+};
+
+&wmac {
+	status = "okay";
+
+	mtd-cal-data = <&art 0x11000>;
+};
+
+&uart {
+	status = "okay";
+};
+
+&pll {
+	clocks = <&clock40mhz>;
+};
+
+&usb {
+	status = "okay";
+
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	hub_port: port@1 {
+		reg = <1>;
+		#trigger-source-cells = <0>;
+	};
+};
+
+&usb_phy {
+	status = "okay";
+};
+
+&mdio0 {
+	status = "okay";
+};
+
+&mdio1 {
+	status = "okay";
+};
```

Comment:
I'd prefer to move the mdio nodes up so they are together with their corresponding ethX node.